analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs

```diff
@@ -82,7 +82,7 @@ public void HasMatchingScope_SingleDiagnostic_WithOneOrMoreScopes_Scanner(bool e
     [DataRow(false, ProjectType.Test, MainTag, MainTag)]
     public void HasMatchingScope_MultipleDiagnostics_WithSingleScope_SonarLint(bool expectedResult, ProjectType projectType, params string[] rulesTag)
     {
-        var diagnostics = rulesTag.Select(x => AnalysisScaffolding.CreateDescriptor(DummyID, x));
+        var diagnostics = rulesTag.Select(x => AnalysisScaffolding.CreateDescriptor(DummyID, x)).ToImmutableArray();
```

Comment:
Can't we make `AnalysisScaffolding.CreateDescriptor` to create an `ImmutableArray` from the start?

Reply:
No, it creates just one. It's the `ruleTag.Select` that makes it a collection